qcodes/instrument/visa.py

```diff
@@ -209,8 +209,11 @@ def write_raw(self, cmd: str) -> None:
         """
         with DelayedKeyboardInterrupt():
             self.visa_log.debug(f"Writing: {cmd}")
-            nr_bytes_written, ret_code = self.visa_handle.write(cmd)
-            self.check_error(ret_code)
+            response = self.visa_handle.write(cmd)
+
+            if isinstance(response, tuple) and len(response) == 2:
+                ret_code = response[1]
+                self.check_error(ret_code)
```

Comment:
I think we should just drop this code block. Evidently this was never meant to return the status code

Reply:
return - no, but check the error and raise a VisaIOError on it - yes, right? or am i misunderstanding something?

Reply:
@astafan8 I mean pyvisa was never meant to return the error code to here

Reply:
yes, thanks (also clarified this in an offline conversation)

Reply:
okay great, I was wondering this as well.  I wasn't exactly sure if the `return_handler` took care of this so it's great for the confirmation.